### PR TITLE
[RND-67] move from Joi to JSON schema

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/ApiPropertyMappingEnhancer.ts [hunk 1]
```diff
@@ -132,6 +151,8 @@ function buildApiPropertyMapping(property: EntityProperty): ApiPropertyMapping {
     isDescriptorCollection,
     descriptorCollectionName: isDescriptorCollection ? apiDescriptorReferenceName(property as DescriptorProperty) : '',
     isCommonCollection,
+    isScalarCommon,
+    isScalarReference,
```
Comment (bradbanister) [inline Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/ApiPropertyMappingEnhancer.ts:155]: Adding these simplifies the schema code

File: Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JsonSchemaEnhancer.ts [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+    description: {
+      type: 'string',
+      description: 'The descriptor description',
+    },
+  },
+  additionalProperties: false,
+  required: ['namespace', 'codeValue', 'shortDescription'],
+};
+
+// SchoolYearEnumeration is a hardcoded exception in the ODS/API
```
Comment (stephenfuqua) [inline Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JsonSchemaEnhancer.ts:65]: This is our secret easter egg for anyone outside of Ed-Fi who tries to use MetaEd & Meadowlark to stand up an alternate data standard  & API without paying attention.

Comment (bradbanister) [reply Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JsonSchemaEnhancer.ts:65]: Maybe I should Base64 encode the comment? 🤣🤣

**Review by stephenfuqua — COMMENTED** [review]
Code looks good. I'm going to run a timing test with the bulk loader before and after this update, just to see what impact, if any, the JSON validator has on the overall application performance.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**